app/lib/features/activities/pages/activities_page.dart

```diff
@@ -20,110 +21,152 @@ class ActivitiesPage extends ConsumerWidget {
+            title: Text('Syncing with your homeserver'),
+            subtitle:
+                Text('This might take a while if you have a large account'),
+          ),
+        ),
+      );
+    } else if (errorMsg != null) {
+      return SliverToBoxAdapter(
+        child: Card(
+          child: ListTile(
+            leading: const Icon(Atlas.warning),
+            title: Text('Error syncing: $errorMsg'),
+            subtitle: Text(
+              retryDuration == null
+                  ? 'retrying ...'
+                  : 'Will retry in ${retryDuration.inMinutes.remainder(60).toString().padLeft(2, '0')}:${retryDuration.inSeconds.remainder(60).toString().padLeft(2, '0')}',
             ),
           ),
         ),
       );
-      children.add(
-        SliverList(
-          delegate: SliverChildBuilderDelegate(
-            (BuildContext ctx, int index) {
-              return InvitationCard(
-                invitation: invitations[index],
-                avatarColor: Colors.white,
-              );
-            },
-            childCount: invitations.length,
+    }
+    return null;
+  }
+
+  List<Widget>? renderInvitations(BuildContext context, WidgetRef ref) {
+    final invitations = ref.watch(invitationListProvider);
+    if (invitations.isEmpty) {
+      return null;
+    }
+    return [
+      SliverToBoxAdapter(
+        child: Padding(
+          padding: const EdgeInsetsDirectional.symmetric(
+            horizontal: 10,
+            vertical: 5,
           ),
+          child: Text(
+            'Invitations',
+            style: Theme.of(context).textTheme.headlineSmall,
+          ),
+        ),
+      ),
+      SliverList(
+        delegate: SliverChildBuilderDelegate(
+          (BuildContext ctx, int index) {
+            return InvitationCard(
+              invitation: invitations[index],
+              avatarColor: Colors.white,
+            );
+          },
+          childCount: invitations.length,
         ),
+      ),
+    ];
+  }
+
+  Widget? renderSessions(BuildContext context, WidgetRef ref) {
+    final allSessions = ref.watch(unknownSessionsProvider);
+    if (allSessions.error != null) {
+      return SliverToBoxAdapter(
+        child: Text("Couldn't load unverified sessions: ${allSessions.error}"),
       );
+    } else if (!allSessions.hasValue) {
+      // we can ignore
+      return null;
     }
-    if (children.isEmpty) {
-      children.add(
-        const SliverToBoxAdapter(
-          child: Center(
-            heightFactor: 1.5,
-            child: EmptyState(
-              title: 'No Activity for you yet',
-              subtitle:
-                  'Notifies you about important things such as messages , invitations or requests.',
-              image: 'assets/images/empty_activity.svg',
+
+    final sessions =
+        allSessions.value!.where((session) => !session.isVerified()).toList();
+    if (sessions.isEmpty) {
+      return null;
```

Comment:
Could've some placeholder indicating that all sessions verified with green badge ;)

Reply:
Not in this case. This page is meant to show the things you need to take care of. If everything is fine, it shouldn't show (here) but instead be removed to decrease the noise.